x-pack/plugins/infra/public/components/asset_details/tabs/overview/metrics/metrics_charts_section.tsx

```diff
@@ -66,6 +72,29 @@ export const ChartGrid = React.memo(
       [setDateRange]
     );
 
+    const handleFilter = useCallback(
+      ({ data, preventDefault }: FilterArgs) => {
+        if (!isChartClickContextData(data)) {
+          return;
+        }
+
+        const { column, row } = extractTableEntryFromChartClickContextData(data);
+
+        if (!column || !row || column.meta.type !== 'date') {
+          return;
+        }
+
+        const timestamp = row[column.id];
+        const rowInterval = calculateChartRowsTimeInterval(data[0].table.rows, column.id);
+        const from = new Date(timestamp).toISOString();
+        const to = new Date(timestamp + rowInterval).toISOString();
```

Comment:
I wonder if it would be better if we encapsulated all these details in a separate function in `chart_utils` and kept the event handler simpler, just receiving the `from` and `to` attribute values from that function. e.g:

```ts
const {from, to} = getRangeFromPopoverFilter(data) //(random name)
if (!from || !to) {
  return
}
setDateRange({ from, to });
preventDefault();
```
Thoughts?

Reply:
Good point, would keep the component cleaner 👍